---
title: "RNASEq Y321"
output:
  word_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```


```{r}
library(reshape2)
library(data.table)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(plotly)
library(pastecs)
library(gridExtra)
library(readxl)
library(BiocManager)
library(DESeq2)
library(topGO)
library("vsn")
library("hexbin")
library(pheatmap)
library(RColorBrewer)
library(glue)
library(org.Dr.eg.db)
library(AnnotationDbi)
```


```{r}
#library(GOplot)
counts_1 <- read.csv(file="1_counts.csv", row.names=1)
all_counts <- read.csv(file="all_counts.csv")
str(all_counts)
#library()
#if (!requireNamespace("BiocManager", quietly = TRUE))
 #install.packages("BiocManager")
#BiocManager::install("DESeq2", version = "3.8")
#loaded a whole bunch of packages and updated old ones
#library("DESeq2")
as.factor(all_counts$GENE)
str(all_counts)
```

#counts_1 <- read.csv(~Desktop/RNAseq/1_counts.csv, header = T)

all_counts <- read.csv(file="all_counts.csv")
str(all_counts)
#library()
#if (!requireNamespace("BiocManager", quietly = TRUE))
 #install.packages("BiocManager")
#BiocManager::install("DESeq2", version = "3.8")
#loaded a whole bunch of packages and updated old ones
library("DESeq2")
as.factor(all_counts$GENE)
str(all_counts)
#################################################################


```{r}
# did this for practice with ALL genes, not just protein coding
#make it into a matrix from a txt file
countData <- as.matrix(read.table("all_countz.txt", header= TRUE, row.names = 1))
head(countData)
#categorize via GFP or not and apply that by column
colData <- data.frame(condition=ifelse(grepl("countsGFP", colnames(countData)), "GFP", "NEG"))
colData
#create counts matrix
dds1 <- DESeqDataSetFromMatrix(countData, colData, formula(~ condition))
#get rid of everything that has zero or 1 counts
ddss <- dds1[ rowSums(counts(dds1)) >1,]
nrow(ddss)
#################################################################
```


```{r}
#alright let's do the protein coding only ones
countData <- as.matrix(read.table("all_countz.txt", header= TRUE, row.names = 1))
head(countData)
#categorize via GFP or not and apply that by column
colData <- data.frame(condition=ifelse(grepl("countsGFP", colnames(countData)), "GFP", "NEG"))
colData
#create counts matrix
dds1 <- DESeqDataSetFromMatrix(countData, colData, formula(~ condition))
#get rid of everything that has zero or 1 counts
ddss <- dds1[ rowSums(counts(dds1)) >1,]
nrow(ddss)
```


```{r}
##################################################################
head(ddss)
str(ddss)

#install some ish##################################
#if (!requireNamespace("BiocManager", quietly = TRUE))
 #install.packages("BiocManager")
#BiocManager::install("vsn", version = "3.8")
################################################
#practice from the manual
#library("vsn")
#library("hexbin")
lambda <- 10^seq(from = -1, to =2, length =1000)
cts <- matrix(rpois(1000*100, lambda), ncol=100)
meanSdPlot(cts, ranks = FALSE)
##################################################
```


```{r}
#transformations
vsd <- vst(ddss, blind=FALSE)
head(assay(vsd), 20)
#all look pretty damn similar in value after that one
colData(vsd)

rld <- rlog(ddss, blind = FALSE)
head(assay(rld), 20)
#much wider range of data with this one

#library("ggplot2")


#now going to show the effect(s) of transformation, plotting one sample against the other
#first need to estimate size factors to account for sequencing depth then specify normalized=TRUE
#sequencing depth correction is automatically done for vst and rlog
trans.effect <- estimateSizeFactors(ddss)

df <- bind_rows(
 as_data_frame(log2(counts(trans.effect, normalized=TRUE)[, 1:2]+1)) %>%
   mutate(transformation = "log2(x+1)"),
 as_data_frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"),
 as_data_frame(assay(rld)[, 1:2]) %>% mutate(transformation = "rlog"))
colnames(df)[1:2] <- c("x", "y")
ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) +
 coord_fixed() + facet_grid( . ~ transformation)
```


```{r}
#####################################################
#SAMPLE DISTANCES
sampledists <- dist(t(assay(vsd)))
sampledists

#using the rlog transformed values
sampledistmatrix <- as.matrix(sampledists)
rownames(sampledistmatrix) <- paste( vsd$condition, sep= "-")
colnames(sampledistmatrix) <- NULL
colors <- colorRampPalette(rev(brewer.pal(9, "Reds")))(255)
pheatmap(sampledistmatrix, clustering_distance_rows = sampledists, clustering_distance_cols = sampledists,
        col=colors)

sampledistmatrix <- as.matrix(sampledists)
rownames(sampledistmatrix) <- paste( rld$condition, sep= "-")
colnames(sampledistmatrix) <- NULL
colors <- colorRampPalette(rev(brewer.pal(9, "Reds")))(255)
pheatmap(sampledistmatrix, clustering_distance_rows = sampledists, clustering_distance_cols = sampledists,
        col=colors)
```


```{r}
#now using poisson distance with the raw counts
library(PoiClaClu)
poisd <- PoissonDistance(t(counts(trans.effect)))
head(poisd)
sample_pois_matrix <- as.matrix(poisd$dd)
rownames(sample_pois_matrix) <- paste(trans.effect$condition)
colnames(sample_pois_matrix) <- NULL
pheatmap(sample_pois_matrix,
        clustering_distance_rows = poisd$dd,
        clustering_distance_cols = poisd$dd,
        col=colors)
```


```{r}
#TIME FOR SOME PCA
#PCA with vst transformation

plotPCA(vsd, ntop=20000) + theme_classic()


PCAdata_vsd <- plotPCA(vsd, returnData = TRUE)
PCAdata_vsd

#PCA with rlog transformation
plotPCA(rld)
PCAdata_rld <- plotPCA(rld, returnData = TRUE)
PCAdata_rld
#negative sample #6 looks way different... #5 and #4 are from the same day, this one
#was from the last sort I did....

#NOW FINALLY TIME FOR SOME DGE RESULTS :)
seek <- DESeq(ddss)
res <- results(seek)
mcols(res, use.names=TRUE)
summary(res)

#lower the FDR threshold
res0.5 <- results(seek, alpha=0.5)
table(res0.5$padj < 0.05)

#raise the log2FC threshold to test for genes that show more
#substantial changes due to condition
resLFC1 <- results(seek, lfcThreshold = 1)
table(resLFC1$padj < 0.01)
```


```{r}
#before we play around with graphs, let's annotate then export the results into a csv
library(AnnotationDbi)
library(org.Dr.eg.db) #this is the HUMAN one I want the ZF one... trying one by same author for ZF
#if (!requireNamespace("BiocManager", quietly = TRUE))
 #install.packages("BiocManager")
#BiocManager::install()
```


```{r}
library(org.Dr.eg.db)
#columns(org.Hs.eg.db)
#BiocManager::install("org.Dr.eg.db")

res$symbol <- mapIds(org.Dr.eg.db,
                    keys=row.names(res),
                    column="SYMBOL",
                    keytype = "ENSEMBL",
                    multiVals = "first")
res$entrez <- mapIds(org.Dr.eg.db,
                    keys=row.names(res),
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals = "first")
resOrder <- res[order(res$pvalue),]
head(resOrder)
```


```{r}
#getwd()
write.csv(resOrder,file="DEGseq_RawResults_3-6-19.csv")
#vignette(DEseq2)
#"DESeq2"
```


```{r}
#########################################################
#doing everything sans group six since it looks like an outlier
library(DESeq2)
library(stringi)
countData2 <- as.matrix(read.table("1-5_PC_counts.txt", header= TRUE, row.names = 1))
head(countData2)
#categorize via GFP or not and apply that by column
colData2 <- data.frame(condition=ifelse(grepl("countsGFP", colnames(countData2)), "GFP", "NEG"))
colData2
```


```{r}
#create counts matrix
dds2 <- DESeqDataSetFromMatrix(countData2, colData2, formula(~ condition))
#get rid of everything that has zero or 1 counts
ddss2 <- dds2[ rowSums(counts(dds2)) >1,]
nrow(ddss2)

#from this you can see that there are 22999 genes or factors for the total dataset.
```


```{r}
vsd2 <- vst(ddss2, blind=FALSE)
head(assay(vsd2), 20)
#all look pretty damn similar in value after that one
colData2 <- data.frame(condition=ifelse(grepl("countsGFP", colnames(countData2)), "GFP", "NEG"))
#colData2(vsd2)

rld2 <- rlog(ddss2, blind = FALSE)
head(assay(rld2), 20)

#pheatmat using the rlog trans for samples 1-5
library(pheatmap)
library(RColorBrewer)
sampledists2rld <- dist(t(assay(rld2)))
sampledists2rld
sampledistmatrix2rld <- as.matrix(sampledists2rld)
rownames(sampledistmatrix2rld) <- paste( vsd2$condition, sep= "-")
colnames(sampledistmatrix2rld) <- NULL
colors2rld <- colorRampPalette(rev(brewer.pal(9, "BuPu")))(255)
pheatmap(sampledistmatrix2rld, clustering_distance_rows = sampledists2rld, clustering_distance_cols = sampledists2rld,
        col=colors2rld)

sampledistmatrix <- as.matrix(sampledists)
rownames(sampledistmatrix) <- paste( rld2$condition, sep= "-")
colnames(sampledistmatrix) <- NULL
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
pheatmap(sampledistmatrix, clustering_distance_rows = sampledists, clustering_distance_cols = sampledists,
        col=colors)
```


```{r}
#PCA with vsd trans
plotPCA(vsd2)
PCAdata_vsd2 <- plotPCA(vsd2, returnData = TRUE)
PCAdata_vsd2

#PCA with rlog transformation
plotPCA(rld2)
PCAdata_rld2 <- plotPCA(rld2, returnData = TRUE)
PCAdata_rld2
```


```{r}
#poisson distances with 1-5
library(PoiClaClu)
trans.effect2 <- estimateSizeFactors(ddss2)
poisd2 <- PoissonDistance(t(counts(trans.effect2)))
head(poisd)
sample_pois_matrix2 <- as.matrix(poisd2$dd)
rownames(sample_pois_matrix2) <- paste(trans.effect2$condition)
colnames(sample_pois_matrix2) <- NULL
pheatmap(sample_pois_matrix2,
        clustering_distance_rows = poisd2$dd,
        clustering_distance_cols = poisd2$dd,
        col=colors2rld)
poisd2$dd
```


```{r}
#not sure if i did this right so trying it from scratch
poisd3 <- PoissonDistance(t(counts(ddss2)))
head(poisd)
sample_pois_matrix3 <- as.matrix(poisd3$dd)
rownames(sample_pois_matrix3) <- paste(ddss2$condition)
colnames(sample_pois_matrix3) <- NULL
pheatmap(sample_pois_matrix3,
        clustering_distance_rows = poisd3$dd,
        clustering_distance_cols = poisd3$dd,
        col=colors2rld)
#worked the same way so its fine
```


```{r}
###deseq
seek2 <- DESeq(ddss2)
res2 <- results(seek2)
mcols(res2, use.names=TRUE)
#THISIS WHERE YOU CAN SEE WHAT GROUP WAS COMPARED TO WHICH
summary(res2)
colnames(res2)

#give those ensembl IDs some gene names
res2$symbol <- mapIds(org.Dr.eg.db,
                    keys=row.names(res2),
                    column="SYMBOL",
                    keytype = "ENSEMBL",
                    multiVals = "first")
res2$entrez <- mapIds(org.Dr.eg.db,
                    keys=row.names(res2),
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals = "first")
resOrder2 <- res2[order(res2$pvalue),]
head(resOrder2)
write.csv(resOrder2, file="DEseq_1to5_3-12-19.csv")

```


```{r}
########################################################
#now lets try some MA (mean of normalized counts) plots
#if (!requireNamespace("BiocManager", quietly = TRUE))
 #install.packages("BiocManager")
#BiocManager::install("apeglm", version = "3.8")
library(apeglm)
#MA plot for 1-6
resultsNames(seek)
MA_res <- lfcShrink(seek, coef="condition_NEG_vs_GFP", type="apeglm")
plotMA(MA_res, ylim = c(-10,10), main="MA Plot for All Samples")
```


```{r}
#MA plot for 1-5
MA_res2 <- lfcShrink(seek2, coef="condition_NEG_vs_GFP", type="apeglm")
plotMA (MA_res2, ylim = c(-10, 10), main = "Mean-Difference Plot", alpha= 0.05)
#alpha = defines the p-value threshold
plotMA(MA_res2, ylim = c(-10,10), main = "Mean Difference plot")
resultsNames(seek2)
#shrinking via Bayesian procedure to moderate ("SHRINK") log2 FC
#from genes with very low counts and highly variable counts via ;fcshrink
library(ggpubr)
library(BiocGenerics)
```







```{r}
#Top gene expressed there is lhx8
plotMA(MA_res2, ylim = c(-10,10))
topGene <- rownames(res)[which.min(res$padj)]
with(res[topGene, ], {
  points(baseMean, log2FoldChange, col="dodgerblue", cex=2, lwd=2)
  text(baseMean, log2FoldChange, topGene, pos=2, col="dodgerblue")
})

#Bottom Gene expressed here in pex6: peroxisomal biogenesis gene
plotMA(MA_res2, ylim = c(-10,10))
bottomGene <- rownames(res)[which.max(res$padj)]
with(res[bottomGene, ], {
  points(baseMean, log2FoldChange, col="red", cex=2, lwd=2)
  text(baseMean, log2FoldChange, bottomGene, pos=2, col="red")
})
```


```{r}
#This is a different type of plot
ggmaplot(MA_res2, main = expression("NEG" %->% "GFP"),
   fdr = 0.01, fc = 2, size = 0.4,
   palette = c("#B31B21", "#1465AC", "darkgray"),
   genenames = as.vector(MA_res2$name),
   legend = "top", top = 5,
   font.label = c("bold", 11),
   font.legend = "bold",
   font.main = "bold",
   ggtheme = ggplot2::theme_classic())
```


```{r}
#MA_res2 <- lfcShrink(seek2, coef="condition_NEG_vs_GFP", type="apeglm")
#plotMA((MA_res2, ylim = c(-10,10), main="Mean-Difference Plot", alpha = 0.05), label.rectangle = TRUE)
#alpha = defines the p-value threshold

#MA if we didn't shink the values..
res.noshr <- results(seek2, name="condition_NEG_vs_GFP")
plotMA(res.noshr, ylim = c(-10, 10))
```



```{r}
plotMA(MA_res2, ylim=c(-5,5))
with(res[topGene, ], {
  points(baseMean, log2FoldChange, col="red", cex=2, lwd=2)
  text(baseMean, log2FoldChange, topGene, pos=2, col="red")
})
```
#Making Go Plots and Volcano plots
 
```{r}
library(clusterProfiler)
library(GO.db)
library(GOplot)
library(org.Dr.eg.db)
library(BiocManager)
library(Matrix)
library(tidyverse)
library(ggplot2)
```


```{r}
#Cluster <- read.csv("~/Desktop/DATA/cluster19_CVvGF.csv")
Clus1 <- read_csv("~/Desktop/RNAseq/DEseq_1to5_3-12-19.csv")

colnames(Clus1)
head(Clus1)
colnames(Clus1) <- c( "ENSEMBL", "baseMean", "log2FoldChange", "lfcSE", "stat", "pvalue", "padj", "symbol", "entrez")
Clus1_sub_GFPup <- subset(Clus1, pvalue<.05)
#Y321 <- write.table(res2p, "GOresults.tsv", sep="\t", col.names=TRUE)
#Y321, 
#read_table(Y321)
```
#Showing genes that are upregulated in GFP cells and the GO terms for the same. 

```{r}
library(doParallel)
Clus1_sub_GFPup <- subset(Clus1, pvalue<.05)
#note that this is for the pvalue. 
go_Clus1_GFPup <- enrichGO(gene         = Clus1_sub_GFPup$ENSEMBL,
                       OrgDb         = org.Dr.eg.db,
                       keyType       = 'ENSEMBL',
                       ont           = "ALL",
                       pAdjustMethod = "none",
                       pvalueCutoff  = 0.05,
                       qvalueCutoff  = 0.05)

go_Clus1_GFPup

barplot(go_Clus1_GFPup, showCategory=80,  title="Cluster 1: \nPathways Upregulated within GFP Cells \npval cutoff at .05")
Clus1_resultsGFP <- go_Clus1_GFPup@result
write.table(Clus1_resultsGFP, "Clus1_resultsGFP.tsv", sep="\t", col.names=TRUE)
#Cluster1_ <- go_Cluster1@result
```


```{r}
#setting pvalue cutoff at 0.01
go_Cluster_GFP_fdr <- enrichGO(gene         = Clus1_sub_GFPup$ENSEMBL,
                       OrgDb         = org.Dr.eg.db,
                       keyType       = 'ENSEMBL',
                       ont           = "ALL",
                       pAdjustMethod = "fdr",
                       pvalueCutoff  = 0.001,
                       qvalueCutoff  = 0.001)
barplot(go_Cluster_GFP_fdr, showCategory=30,  title="Cluster x: \nPathways Upregulated within GFP Cells")

##Cluster_resultsGFP_fdr <- go_Cluster_GFP_fdr@result
#write.table(Cluster_resultsGFP_fdr, "GOresults_GFPup_fdr.tsv", sep="\t", col.names=TRUE)
#write.table(Cluster_results1, "GOresults_UP.tsv", sep="\t", col.names=TRUE)
```
#Some GO plots based on Log2FC

```{r}
#What happens if we change the level of log2FC
Clus1_sub_GFPFC <- subset(Clus1, log2FoldChange<0)

go_Cluster_GFPFC <- enrichGO(gene         = Clus1_sub_GFPFC$ENSEMBL,
                       OrgDb         = org.Dr.eg.db,
                       keyType       = 'ENSEMBL',
                       ont           = "ALL",
                       pAdjustMethod = "none",
                       pvalueCutoff  = 0.05,
                       qvalueCutoff  = 0.05)
barplot(go_Cluster_GFPFC, showCategory=100,  title="Cluster x: \nPathways Upregulated within GFP Cells \npval cutoff at .05")

Cluster_resultsGFPFC <- go_Cluster_GFPFC@result
write.table(Cluster_resultsGFPFC, "GOresults_GFPup_FC.tsv", sep="\t", col.names=TRUE)

```


#Try to use padj values
```{r}
Clus1_sub_GFP_pad <- subset(Clus1, padj<0.05)

go_Cluster_GFP_1 <- enrichGO(gene         = Clus1_sub_GFP_pad$ENSEMBL,
                       OrgDb         = org.Dr.eg.db,
                       keyType       = 'ENSEMBL',
                       ont           = "ALL",
                       pAdjustMethod = "none",
                       pvalueCutoff  = 0.05,
                       qvalueCutoff  = 0.05)
barplot(go_Cluster_GFP_1, showCategory=30,  title="Cluster x: \nPathways Upregulated within GFP Cells \npadjusted cutoff at .05")
#(go_Cluster_GFP_1, showCategory=30,  title="Cluster x: \nPathways Upregulated within GFP Cells \npadjusted cutoff at .05")
Cluster_resultsGFP_1 <- go_Cluster_GFP_1@result
write.table(Cluster_resultsGFP_1, "GOresults_GFPup_Pad.tsv", sep="\t", col.names=TRUE)
```


```{r}
Clus1_sub_GFP_pad <- subset(Clus1, padj>0.05)

go_Cluster_GFP_2<- enrichGO(gene         = Clus1_sub_GFP_pad$ENSEMBL,
                       OrgDb         = org.Dr.eg.db,
                       keyType       = 'ENSEMBL',
                       ont           = "ALL",
                       pAdjustMethod = "none",
                       pvalueCutoff  = 0.05,
                       qvalueCutoff  = 0.05)
barplot(go_Cluster_GFP_2, showCategory=30,  title="Cluster x: \nPathways Upregulated within GFP Cells \npadjusted cutoff at .05")
#(go_Cluster_GFP_1, showCategory=30,  title="Cluster x: \nPathways Upregulated within GFP Cells \npadjusted cutoff at .05")
Cluster_resultsGFP_1 <- go_Cluster_GFP_1@result
write.table(Cluster_resultsGFP_2, "GOresults_GFPup_Pad.tsv", sep="\t", col.names=TRUE)
```


```{r}
library(GOplot)
library(topGO)
library(PANTHER.db)
#library(RDAVIDWebService)


#GOCircle(go_Cluster_GFP_fdr$p.adjust)
```




```{r}
ggmaplot(MA_res2, main = expression("NEG" %->% "GFP"),
   fdr = 0.05, fc = 2, size = 0.4,alpha = 10,
   palette = c("#B31B21", "#1465AC", "darkgray"),
   genenames = as.vector(MA_res2$name),
   legend = "top", top = 10,
   font.label = c("bold", 11), label.rectangle = TRUE,
   font.legend = "bold",
   font.main = "bold",
   ggtheme = ggplot2::theme_classic())
```

```{r}

ggmaplot(MA_res2, main = expression("NEG" %->% "GFP"),
         fdr = 0.05, fc = 2, size = 0.4,
         genenames = as.vector(MA_res$name),
         ggtheme = ggplot2::theme_classic(),
         top = 5, label.select = c("ENSDARG00000004023", "ENSDARG00000002330", "ENSDARG00000006896")
)

colnames(MA_res2)


```
```{r}
seek2

MA1 <- replace(MA_res2, 1, "lhx8a")
MA_res2
```


```{r}
ggmaplot(MA1, main = expression("NEG" %->% "GFP"),
         fdr = 0.5, fc = 2, size = 3,
         genenames = as.vector(MA1$name),
         ggtheme = ggplot2::theme_classic(),
         top = 1, label.select = c("gal4_Forward", "lhx8a", "UAS_EGFP_Forward")
)

```



```{r}
#now going to make a heatmap
#library(genefilter)
topVarGenes <- order(-rowVars(assay(rld)))[0:1000]
topVarGenes2 <-  head(order(rowVars(assay(rld2)), decreasing=TRUE), 20)
mat2 <- assay(rld2)[topVarGenes2, ]
mat2 <- mat2 - rowMeans(mat2)
anno <- as.data.frame(colData(rld2))
pheatmap(mat2, annotation_col = anno)


#There are forebrain gene but not annotated 
#going to make one with the raw counts
topVarGenes <- head(order(-rowVars(assay(rld2))),50)
mat <- assay(rld2)[ topVarGenes2, ]
mat <- mat - rowMeans(mat)
dfz <- as.data.frame(colData(rld2)[,c("condition", "sizeFactor")])
pheatmap(mat, fontsize_number = 0.8 * fontsize, annotation_col=dfz)

```


```{r}
#going to make a heatmap of the counts and annotated genes
#ANNOTATED  forebrain GENESSSS...
countData4 <- as.matrix(read.csv("forebrain_counts_annotated.csv", header= TRUE, row.names=1))
head(countData4)
## getting the code here to make annotated heatmap

#categorize via GFP or not and apply that by column
colData4 <- data.frame(condition=ifelse(grepl("countsGFP", colnames(countData3)), "GFP", "NEG"))
colData4
#create counts matrix
dds4 <- DESeqDataSetFromMatrix(countData4, colData4, formula(~ condition))
#get rid of everything that has zero or 1 counts
ddss4 <- dds4[ rowSums(counts(dds4)) >1,]
#nrow(ddss3)<- vst(ddss3, blind=FALSE)
#head(assay(vsd3), 20)

dds3 <- DESeqDataSetFromMatrix(countData3, colData3, formula(~ condition))
#get rid of everything that has zero or 1 counts
ddss3 <- dds3[ rowSums(counts(dds3)) >1,]
nrow(ddss3)
head(ddss3)

#all look pretty damn similar in value after that one
colData(vsd2)
rld4 <- rlog(dds3, blind = FALSE)
head(assay(rld3), 20)

topVarGenes4 <-  head(order(rowVars(assay(rld4)), decreasing=TRUE), 20)
mat4 <- assay(rld4)[topVarGenes4, ]
mat4 <- mat4 - rowMeans(mat4)
anno4 <- as.data.frame(colData(rld4))

#dfz4 <- as.data.frame(colData(rld4)[("condition", "sizeFactor")])
fz4 <- as.data.frame(colData(rld4)["condition"])
pheatmap(mat4, fontsize_number = 2 * fontsize, annotation_col=fz4)
pheatmap(mat4, fontsize_number = 0.8 * fontsize, annotation_col=fz4)

#Top 20
pheatmap(mat4, fontsize_number = 20, annotation_col=fz4,
        color = colorRampPalette(rev(brewer.pal(n = 10, name = "RdYlBu")))(20),
        cellwidth =20, fontsize_row =10, fontsize_col = 10, angle_col = 45)
#pheatmap(mat3, annotation_col = anno3)



```


```{r}
#LETS TRY THIS GOD FORSAKEN HEATMAP
#first picking out the first # of genes that are most variably expressed
#top 20 overall
topVarGenes3 <-  head(order(rowVars(assay(rld3)), decreasing=TRUE), 20)
mat3 <- assay(rld3)[topVarGenes3, ]
mat3 <- mat3 - rowMeans(mat3)
anno3 <- as.data.frame(colData(rld3))

#dfz3 <- as.data.frame(colData(rld3)[("condition", "sizeFactor")])
fz3 <- as.data.frame(colData(rld3)["condition"])
pheatmap(mat3, fontsize_number = 2 * fontsize, annotation_col=fz3)
pheatmap(mat3, fontsize_number = 0.8 * fontsize, annotation_col=fz3)

#Top 20
pheatmap(mat3, fontsize_number = 20, annotation_col=fz3,
        color = colorRampPalette(rev(brewer.pal(n = 10, name = "RdYlBu")))(20),
        cellwidth =20, fontsize_row =10, fontsize_col = 10, angle_col = 45)
pheatmap(mat3, annotation_col = anno3)
```
```{r}
# BiocManager::install('EnhancedVolcano')
 library(EnhancedVolcano) 
library(volcano3D)
library(magrittr)
```


`





```{r}

EnhancedVolcano(resOrder2,
    lab = rownames(resOrder2),
    x = 'log2FoldChange',
    y = 'pvalue')+
theme_classic()
```





```{r}
  EnhancedVolcano(resOrder2,
    lab = rownames(resOrder2),
    x = 'log2FoldChange',
    y = 'pvalue',
    title = 'GFP versus Non-GFP',
    #pCutoff = 10e-32,
    FCcutoff = 2,
    pointSize = 3.0,
    labSize = 6.0)
```


```{r}

#library(ggplot2)
V <- ggplot(data = MA_res2, 
            aes(x = log2FoldChange, 
                y = -log10(x$FDR), 
                colour=expression,
                label = x$label)) +
  geom_point(alpha=0.4, size=3.5) +
  scale_color_manual(values=c("blue", "grey","red"))+
  xlim(c(-4.5, 4.5)) +
  geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +
  geom_hline(yintercept = 1.301,lty=4,col="black",lwd=0.8) +
  labs(x="log2(fold change)",
       y="-log10 (adj.p-value)",
       title="Differential expression")  +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), 
        legend.position="right", 
        legend.title = element_blank())
V

```

```{r}
#NB: the relative up and down regulation is in relation to GFP negative samples: dots in blue symbolize downregulation in GFP negative. 
ggmaplot(MA_res2, main = expression("GFP" %->% "NEG"),  
         fdr = 0.05, fc = 2, size = 1,
         palette = c("#B31B21", "#1465AC", "darkgray"),
         genenames = as.vector(MA_res2$name),
         font.label = c("bold", 9), label.rectangle = F,
         label.arrow = TRUE,
         font.legend = "bold",
         font.main = "bold",
         ggtheme = ggplot2::theme_classic(),
         top = 3, label.select = c("ENSDARG00000015537", "ENSDARG00000094809")
)
```




